src/Control/Distributed/Closure.hs

```diff
@@ -59,6 +61,13 @@ import Data.Constraint (Dict(..))
 class c => Static c where
   closureDict :: Closure (Dict c)
 
+instance (Static c1, Static c2, Typeable c1, Typeable c2, (c1, c2)) => Static (c1, c2) where
+  closureDict = static pairDict `cap` closureDict `cap` closureDict
+
+-- Needs to be defined at top-level for GHC <8.4 compat.
+pairDict :: Dict c1 -> Dict c2 -> Dict (c1, c2)
+pairDict Dict Dict = Dict
```

Comment:
That together with `Dict ()` make `Dict` a cartesian functor. I think this should be, ultimately, contributed up to the constraints package.

Reply:
The `constraints` package already has something similar, in the form of `(&&&)`. But using it here is more trouble than it's worth, because need to specialize its type anyways, and then pack/unpack the `Sub` constructrions.